[ELY-2360] Change OIDC_STATE delimiter

cboehme opened the pull request:
Change the delimiter character sequence used in the OIDC_STATE cookie
to avoid clashes when the same character sequence appears in the encoded
tokens stored in the cookie.

Additionally, a debug log message is added to print the value of the
OIDC_STATE cookie if the value could not be split into its components.

https://issues.redhat.com/browse/ELY-2360

cboehme committed:
[ELY-2360] Change OIDC_STATE delimiter

Change the delimiter character sequence used in the OIDC_STATE cookie
to avoid clashes when the same character sequence appears in the encoded
tokens stored in the cookie.

Additionally, a debug log message is added to print the value of the
OIDC_STATE cookie if the value could not be split into its components.

fjuma commented on http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcCookieTokenStore.java:
```diff
@@ -206,7 +206,8 @@ public static OidcPrincipal<RefreshableOidcSecurityContext> getPrincipalFromCook
         String[] tokens = cookieVal.split(DELIM);
         if (tokens.length != EXPECTED_NUM_TOKENS) {
-            log.warnf("Invalid format of %s cookie. Count of tokens: %s, expected 3", OIDC_STATE_COOKIE, tokens.length);
+            log.warnf("Invalid format of %s cookie. Count of tokens: %s, expected %s", OIDC_STATE_COOKIE, tokens.length, EXPECTED_NUM_TOKENS);
```
@cboehme Thanks for working on this! I think it would be good to update this so that if the new delimiter isn't found, then we fallback to trying the old delimiter. Just want to make sure that the old delimiter can be handled properly so old cookies aren't considered invalid.

cboehme replied:
Good point. I added a fall back to the old delimiter.

cboehme committed:
[ELY-2360] Accept cookies containing old delimiter

Commit 3a654474cb52194f64512bd75a251cc0b72d8302 changed the delimiter
used to separate tokens in cookies. This change renders existing cookies
which still use the old delimiters invalid, though. To avoid this, a
fallback for the old delimiter is added.

fjuma reviewed (approved):
Thanks very much @cboehme!
